tests/framework/helpers/UrlTest.php

```diff
@@ -121,15 +121,25 @@ public function testToRoute()
     public function testCurrent()
     {
         $this->mockAction('page', 'view', null, []);
-        Yii::$app->request->setQueryParams(['id' => 10, 'name' => 'test', 10 => 0]);
-
-        $this->assertEquals('/base/index.php?r=page%2Fview&id=10&name=test&10=0', Url::current());
-        $this->assertEquals('/base/index.php?r=page%2Fview&id=20&name=test&10=0', Url::current(['id' => 20]));
-        $this->assertEquals('/base/index.php?r=page%2Fview&name=test&10=0', Url::current(['id' => null]));
-        $this->assertEquals('/base/index.php?r=page%2Fview&name=test&10=0&1=yes', Url::current(['id' => [], 1 => 'yes']));
-        $this->assertEquals('/base/index.php?r=page%2Fview&name=test&10=0', Url::current(['id' => []]));
-        $this->assertEquals('/base/index.php?r=page%2Fview&name=test', Url::current(['id' => null, 10 => null]));
-        $this->assertEquals('/base/index.php?r=page%2Fview&name=test&1=yes', Url::current(['id' => null, 10 => null, 1 => 'yes']));
+        $params = ['id' => 10, 'name' => 'test', 10 => 0, 'arr' => ['attr_one' => 1, 'attr_two' => 2]];
+        Yii::$app->request->setQueryParams($params);
+
+        $uri = '/base/index.php?r=page%2Fview&';
+        $this->assertEquals($uri . 'id=10&name=test&10=0&arr%5Battr_one%5D=1&arr%5Battr_two%5D=2', Url::current());
+        $this->assertEquals($uri . 'id=20&name=test&10=0&arr%5Battr_one%5D=1&arr%5Battr_two%5D=2',
+            Url::current(['id' => 20]));
+        $this->assertEquals($uri . 'name=test&10=0&arr%5Battr_one%5D=1&arr%5Battr_two%5D=2',
+            Url::current(['id' => null]));
+        $this->assertEquals($uri . 'name=test&10=0&arr%5Battr_one%5D=1&arr%5Battr_two%5D=2&1=yes',
+            Url::current(['id' => [], 1 => 'yes']));
+        $this->assertEquals($uri . 'name=test&10=0&arr%5Battr_one%5D=1&arr%5Battr_two%5D=2',
+            Url::current(['id' => []]));
+        $this->assertEquals($uri . 'name=test&arr%5Battr_one%5D=1&arr%5Battr_two%5D=2',
+            Url::current(['id' => null, 10 => null]));
+        $this->assertEquals($uri . 'name=test&arr%5Battr_one%5D=1&arr%5Battr_two%5D=2&1=yes',
+            Url::current(['id' => null, 10 => null, 1 => 'yes']));
+        $this->assertEquals($uri . 'arr%5Battr_two%5D=2&1=yes',
+            Url::current(['arr' => ['attr_one' => null], 'id' => null, 'name' => null, 10 => null, 1 => 'yes']));
```

Comment:
Please add new assertions for arrays instead of modifying all existing - it kills readability of this test...

Reply:
@rob006 , I think it's look better now